### ISSUE TITLE
React style error when using inline styles

### ISSUE THREAD

**Issue by ThomWright** [GitbookIO/gitbook#824]
I was about to try to align my `img` tag using inline styles, when the following error appeared in the preview:

**Source**
```html
<img src="./simple-3-node-circuit.jpg" style="">
```

![image](https://cloud.githubusercontent.com/assets/1884486/8674821/37677afc-2a38-11e5-9496-bebfe585beab.png)

**Comment by zning1994** [comment]
You can delete the "style="" "and try again.

**Comment by ThomWright** [comment]
Well, sure, that stops the error, but I'm trying to use inline styles to horizontally centre an image. E.g.

```html
<img src="../imgs/simple-3-node-circuit.jpg"
  style="display: block;
    margin-left: auto;
    margin-right: auto"/>
```

CSS from [here](http://www.w3.org/Style/Examples/007/center.en.html#block).

**Comment by zning1994** [comment]
It seems that markdown couldn't use CSS to adjust the words in this display.

**Comment by znation** [comment]
React uses JS objects to express inline styles, and does not support strings (as shown above). See http://facebook.github.io/react/tips/inline-styles.html

**Comment by ThomWright** [comment]
Sure, but the fact that this is implemented with React shouldn't be exposed to the user.

**Comment by AaronO** [comment]
@ThomWright You're right, React should not be exposed to the user it's an implementation detail.

Just to give you an idea of why this happens, we apply virtual dom diffs when updating the preview pane to improve performance and prevent flickering.

The editor now uses React, in previous versions we used the `virtual-dom` library directly which worked extremely well, but when we switched to React for the rest of the editor it made sense to get the preview updating done via React's diffing library rather than including another `virtual-dom` library.

We'll fix this soon, there are 3 solutions that we should explore:
    * When there's a react error, fallback to a full forced preview refresh
    * Try an use React's lower level virtualdom library (I'm not sure this is possible)
    * Switch back to our previous strategy (and use a separate virtualdom library)

I hope that explains things, if you have any thoughts I would love to hear them !
